apps/entity-service/src/entities/processors/nursery.processor.ts

```diff
@@ -0,0 +1,122 @@
+      } else if (query.sort.field === "projectName") {
+        builder.order(["project", "name", query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field !== "id") {
+        throw new BadRequestException(`Invalid sort field: ${query.sort.field}`);
+      }
+    }
+
+    const frameworkPermissions = permissions
+      ?.filter(name => name.startsWith("framework-"))
+      .map(name => name.substring("framework-".length) as FrameworkKey);
+    if (frameworkPermissions?.length > 0) {
+      builder.where({ frameworkKey: { [Op.in]: frameworkPermissions } });
+    } else if (permissions?.includes("manage-own")) {
+      builder.where({ projectId: { [Op.in]: ProjectUser.userProjectsSubquery(userId) } });
+    } else if (permissions?.includes("projects-manage")) {
+      builder.where({ projectId: { [Op.in]: ProjectUser.projectsManageSubquery(userId) } });
+    }
+
+    for (const term of ["status", "updateRequestStatus", "frameworkKey"]) {
+      if (query[term] != null) builder.where({ [term]: query[term] });
+    }
+
+    if (query.search != null) {
+      builder.where({ name: { [Op.like]: `%${query.search}%` } });
+    }
+
+    if (query.projectUuid != null) {
+      const project = await Project.findOne({ where: { uuid: query.projectUuid }, attributes: ["id"] });
+      if (project == null) {
+        throw new BadRequestException(`Project with uuid ${query.projectUuid} not found`);
+      }
+      builder.where({ projectId: project.id });
+    }
+
+    return { models: await builder.execute(), paginationTotal: await builder.paginationTotal() };
+  }
+
+  async addFullDto(document: DocumentBuilder, nursery: Nursery): Promise<void> {
+    const nuseryId = nursery.id;
+
+    const nurseriesReports = await NurseryReport.nurseries([nuseryId]).findAll({
+      attributes: ["id", "seedlingsYoungTrees"]
+    });
+
+    const seedlingsGrownCount = nurseriesReports.reduce((sum, { seedlingsYoungTrees }) => {
+      return sum + (seedlingsYoungTrees ?? 0);
+    }, 0);
+
+    const nurseryReportsTotal = nurseriesReports.length;
+    const overdueNurseryReportsTotal = await this.getTotalOverdueReports(nuseryId);
+    const props: AdditionalNurseryFullProps = {
+      seedlingsGrownCount,
+      nurseryReportsTotal,
+      overdueNurseryReportsTotal,
+
+      ...(this.entitiesService.mapMediaCollection(
+        await Media.nursery(nuseryId).findAll(),
+        Nursery.MEDIA
+      ) as NurseryMedia)
+    };
+
+    document.addData(nursery.uuid, new NurseryFullDto(nursery, props));
+  }
+
+  async addLightDto(document: DocumentBuilder, nursery: Nursery): Promise<void> {
+    const nuseryId = nursery.id;
+
+    const nurseriesReports = await NurseryReport.nurseries([nuseryId]).findAll({
+      attributes: ["id", "seedlingsYoungTrees"]
+    });
+
+    const seedlingsGrownCount = nurseriesReports.reduce((sum, { seedlingsYoungTrees }) => {
+      return sum + (seedlingsYoungTrees ?? 0);
+    }, 0);
```

Comment:
Same feedback here; use the aggregate query from above instead of fetching all reports.